src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int defaulCapacity = 16;
+    private int capasity;
+    private final float loadFactor = 0.75f;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        capasity = defaulCapacity;
+        size = 0;
+        table = new Node[capasity];
+    }
+
+    static class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        Node(K key, V value,Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > table.length * loadFactor) {
+            resize();
+        }
+        Node<K,V> node = table[hash(key) % table.length];
+        if (node == null) {
+            table[hash(key) % table.length] = new Node<>(key,value,null);
+            size++;
+        } else {
+            while (node != null) {
 
+                if (node.next == null) {
+                    if (node.key == null) {
+                        if (key == null) {
+                            node.value = value;
+                            break;
+                        }
+                    } else if (node.key.equals(key)) {
+                        node.value = value;
+                        break;
+                    }
+                    node.next = new Node<>(key,value,null);
+                    size++;
+                    break;
+                } else if (node.key == null) {
+                    if (key == null) {
+                        node.value = value;
+                        break;
+                    }
+                } else if (node.key.equals(key)) {
+                    node.value = value;
+                    break;
+                }
+                node = node.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> node = table[hash(key) % table.length];
+        if (node != null) {
+            while (node != null) {
+                if (node.key != null) {
+                    if (node.key.equals(key)) {
+                        return node.value;
+                    }
+                } else {
+                    if (key == null) {
+                        return node.value;
+                    }
+                }
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
     }
+
+    private int hash(Object key) {
```

Comment:
The hash function should not depend on the capacity of the table. Usually, hash functions only depend on the key. The modulus operation should be done when you are using the hash to index into the table.